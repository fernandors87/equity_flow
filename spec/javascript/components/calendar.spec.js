import Calendar from 'components/calendar'
import moment from 'moment'

describe('Calendar', () => {

  const date = moment.utc('2017-07-12T00:00:00.000+00:00')

  it('render the component', () => {
    const component = shallow(<Calendar date={ date } />)
    expect(component.html()).toMatchSnapshot()
  })

  it('change calendar to previous month', () => {
    const component = mount(<Calendar date={ date } />)
    component.find('.prev-month').first().simulate('click')
    expect(component.html()).toMatchSnapshot()
  })

  it('change calendar to next month', () => {
    const component = mount(<Calendar date={ date } />)
    component.find('.next-month').first().simulate('click')
    expect(component.html()).toMatchSnapshot()
  })

  it('keep the calendar date state', () => {
    const component = mount(<Calendar date={ date } />)
    component.find('.unselected').first().simulate('click')
    expect(component.state().date).not.toEqualMoment(date)
  })

  it('notify date changes', () => {
    const onDateClick = sinon.spy()
    const component = mount(<Calendar date={ date } onDateClick={ onDateClick } />)
    component.find('.unselected').first().simulate('click')

    expect(onDateClick.callCount).toEqual(1)
    expect(onDateClick.lastCall.args[0]).toEqualMoment(moment.utc('2017-06-25T00:00:00.000+00:00'))
  })

  it('update the state from props', () => {
    const newDate = moment(date).add(1, 'month')
    const newMonth = moment(newDate).startOf('month')
    const component = shallow(<Calendar date={ date } />)
    component.instance().UNSAFE_componentWillReceiveProps({ date: newDate })
    const newState = component.state()
    expect(newState.date).toEqualMoment(newDate)
    expect(newState.month).toEqualMoment(newMonth)
  })
})
