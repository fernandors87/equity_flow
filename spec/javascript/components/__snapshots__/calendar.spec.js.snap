// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar change calendar to next month 1`] = `"<table class=\\"calendar\\"><thead><tr><td><button type=\\"button\\" class=\\"prev-month btn btn-default\\"><span class=\\"glyphicon glyphicon-triangle-left\\"></span></button></td><td colspan=\\"5\\">2017-08</td><td><button type=\\"button\\" class=\\"next-month btn btn-default\\"><span class=\\"glyphicon glyphicon-triangle-right\\"></span></button></td></tr><tr><td>S</td><td>M</td><td>T</td><td>W</td><td>T</td><td>F</td><td>S</td></tr></thead><tbody><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">30</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">31</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">1</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">2</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">3</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">4</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">5</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">6</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">7</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">8</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">9</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">10</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">11</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">12</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">13</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">14</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">15</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">16</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">17</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">18</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">19</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">20</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">21</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">22</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">23</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">24</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">25</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">26</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">27</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">28</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">29</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">30</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">31</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">1</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">2</button></td></tr></tbody></table>"`;

exports[`Calendar change calendar to previous month 1`] = `"<table class=\\"calendar\\"><thead><tr><td><button type=\\"button\\" class=\\"prev-month btn btn-default\\"><span class=\\"glyphicon glyphicon-triangle-left\\"></span></button></td><td colspan=\\"5\\">2017-06</td><td><button type=\\"button\\" class=\\"next-month btn btn-default\\"><span class=\\"glyphicon glyphicon-triangle-right\\"></span></button></td></tr><tr><td>S</td><td>M</td><td>T</td><td>W</td><td>T</td><td>F</td><td>S</td></tr></thead><tbody><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">28</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">29</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">30</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">31</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">1</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">2</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">3</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">4</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">5</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">6</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">7</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">8</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">9</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">10</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">11</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">12</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">13</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">14</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">15</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">16</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">17</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">18</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">19</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">20</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">21</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">22</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">23</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">24</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">25</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">26</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">27</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">28</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">29</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">30</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">1</button></td></tr></tbody></table>"`;

exports[`Calendar render the component 1`] = `"<table class=\\"calendar\\"><thead><tr><td><button type=\\"button\\" class=\\"prev-month btn btn-default\\"><span class=\\"glyphicon glyphicon-triangle-left\\"></span></button></td><td colSpan=\\"5\\">2017-07</td><td><button type=\\"button\\" class=\\"next-month btn btn-default\\"><span class=\\"glyphicon glyphicon-triangle-right\\"></span></button></td></tr><tr><td>S</td><td>M</td><td>T</td><td>W</td><td>T</td><td>F</td><td>S</td></tr></thead><tbody><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">25</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">26</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">27</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">28</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">29</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">30</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">1</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">2</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">3</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">4</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">5</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">6</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">7</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">8</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">9</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">10</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">11</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"selected btn btn-primary\\">12</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">13</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">14</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">15</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">16</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">17</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">18</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">19</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">20</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">21</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">22</button></td></tr><tr><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">23</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">24</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">25</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">26</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">27</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">28</button></td><td class=\\"calendar-day\\"><button type=\\"button\\" class=\\"unselected btn btn-default\\">29</button></td></tr></tbody></table>"`;
